TUTORIAL.md

```diff
@@ -0,0 +1,126 @@
+# Guide to Solving and Reviewing Deli Counter
+
+After running `rspec` you'll see we have seven in total to pass. Let's start with 
+
+### `#line`
+
+```bash
+#line
+    there is nobody in line
+      should say the line is empty (FAILED - 1)
+
+Failures:
+
+  1) Deli Counter #line there is nobody in line should say the line is empty
+     Failure/Error: line(katz_deli)
+     NoMethodError:
+       undefined method `line' for #<RSpec::ExampleGroups::DeliCounter::Line::ThereIsNobodyInLine:0x007fc1f39fdf28>
+```
+To pass this `undefined method 'line'` we have to create a `line` method in our `deli_counter.rb`.
+
+```ruby
+def line
+end
+```
+When we run `rspec --fail-fast`, we'll see the test is still not passing but the error message changed to `wrong number of arguments (1 for 0)`. Our `line` method should accept and `katz_deli` argument.
+
+```ruby
+def line(katz_deli)
+end
+```
+
+Now our method should say `"The line is currently empty."` if there is no one in line.
+
+```ruby
+def line(katz_deli)
+  if katz_deli.length == 0
+    puts "The line is currently empty."
+  end
+end
+```
+Our first test is passing. But we did not tell our method what to do when there are people in line. How are we going to do that. How about we add a `else` statement to our `if` clause.
+
+```ruby
+def line(katz_deli)
+  if katz_deli.length == 0
+    puts "The line is currently empty."
+  else
+    current_line = "The line is currently:"
+    katz_deli.each.with_index(1) do |person, i|
+      current_line << " #{i}. #{person}"
+    end
+    puts current_line
+  end
+end
+```
+Here we are using the `each.with_index(1)` method because we can just pass in the number we want our index to start at. We could have also used the `each_with_index` method, but then we would also have to add 1 to our index because no line starts at 0. Our code in the each block would look like this `current_line << " #{i+1}. #{person}"`. 
```

Comment:
Love this explaination and then showing what it would look like without the argument!

(also didn't know `each_with_index` takes an arg. Cool)